lib/irb/extend-command.rb

```diff
@@ -259,12 +259,34 @@ def self.def_extend_command(cmd_name, cmd_class, load_file, *aliases)
         cmd_class = cmd_class.name
       end
 
-      line = __LINE__; eval %[
-        def #{cmd_name}(*opts, **kwargs, &b)
-          Kernel.require_relative "#{load_file}"
-          ::IRB::ExtendCommand::#{cmd_class}.execute(irb_context, *opts, **kwargs, &b)
-        end
-      ], nil, __FILE__, line
+      # TODO: Remove this condition after 2.0
+      if cmd_name == :irb_measure
+        line = __LINE__; eval %[
+          def irb_measure(type = nil, arg = nil, &block)
+            if block
+              warn <<~MSG
+                Passing a block to measure is deprecated and will be removed in the next IRB major release.
+                Please activate measuring with `IRB.conf[:MEASURE] = true`, and set the callback with `IRB.set_measure_callback(&block)` instead.
+              MSG
+
+              IRB.conf[:MEASURE] = true
+              added = IRB.set_measure_callback(&block)
+              puts "\#{added[0]} is added." if added
+            else
+              Kernel.require_relative "cmd/measure"
+              ::IRB::ExtendCommand::Measure.execute(irb_context, type, arg)
+            end
+          end
+        ], nil, __FILE__, line
```

Comment:
I know this is not pretty,  but this is the only way I can think of to make command refactoring easy while not introducing a n immediate breaking change.